### PR TITLE
Update Go to 1.14.1

### PR DESCRIPTION
**Pull request by tklauser**
Go 1.14.1 was just released: https://groups.google.com/forum/#!topic/golang-dev/cYulRsW466g

Also includes some build system/CI cleanups, see individual commits for details.

### CODE REVIEW

**Comment by tklauser** [comment]
Just noticed that I'll also need to update the `cilium-builder` and `cilium-runtime` images. Will try that tomorrow.

File: .travis.yml.tmpl [hunk 1]
```diff
@@ -12,10 +12,6 @@ arch:
   - amd64
   - arm64
 
-jobs:
```
Comment (aanm) [inline .travis.yml.tmpl:15]: Was this accidentally removed?

Comment (tklauser) [reply .travis.yml.tmpl:15]: No, this is by intention. PR #10540 deleted `allow_failures` from `.travis.yml` but not from `.travis.yml.tmpl`. This leads to `allow_failures` being added again when running `make update-travis-go-version`. Thus I added commit 034f8a9dde79ffaa93f3f6ef43ec3b5501a5042b to this PR.

Comment (aanm) [reply .travis.yml.tmpl:15]: LGTM. We should have a checker that check if the `template` files are out of sync with the generated files. Not blocking for this PR ofc.

**Comment by tklauser** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Review by jrajahalme — APPROVED** [review]

**Pull request merged**